### DISCUSSION TITLE
Sometimes an empty order is added

### DISCUSSION THREAD

**Discussion by de-raaf-media** [duncanmcclean/simple-commerce#1060] [Help]
Sometimes an empty order is added (next to the active order which is still active after the empty order is added). This also happens when the orders is empty and I load the page for the first time but maybe that is how the system works? That an empty order is generated for a new session?

I did not notice this before so it might be a problem on my end. So I don't report this as a bug.

```
id	order_number	order_status	payment_status	items	grand_total	items_total	tax_total	shipping_total	coupon_total	shipping_name	shipping_address	shipping_address_line2	shipping_city	shipping_postal_code	shipping_region	shipping_country	billing_name	billing_address	billing_address_line2	billing_city	billing_postal_code	billing_region	billing_country	use_shipping_address_for_billing	customer_id	coupon	gateway	data	created_at	updated_at
17	1001	cart	unpaid	[]	0	0	0	0	0	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL	NULL	0	NULL	NULL	NULL	{"site": "default"}	2024-04-19 11:41:33	2024-04-19 11:41:33
```

Update: Looks like the Order.php is randomly called. I can't figure out why.

**Comment by de-raaf-media** [comment]
Browsing through the tickets to find anything related to creating orders I stumbled upon issue #647 (Prevent debugbar from creating orders). As it turns out when disabling the debugbar empty orders are no longer created. Ik cannot figure out why this occurs but for me it is good to know that on production empty orders probably will not be created then.

**Reply by duncanmcclean** [reply]
Ah yes, I thought I had sorted that issue with Debugbar but obviously not. Feel free to open a bug report for it.
